File: week01/day07/backend/routers/reminder.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from dependencies import get_db, get_current_worker
from services import reminder as service
from utils.response import ok, ok_page

router = APIRouter(prefix="/api/reminders", tags=["提醒"])


@router.get("")
def list_reminders(
    page: int = Query(1, ge=1),
    pageSize: int = Query(20, ge=1, le=100),
    db: Session = Depends(get_db),
    worker=Depends(get_current_worker),
):
    result = service.list_reminders(db, worker.id, page=page, page_size=pageSize)
    return ok_page(result["data"], result["total"], page, pageSize)


@router.patch("/{reminder_id}/read")
def mark_read(
    reminder_id: int,
    db: Session = Depends(get_db),
    worker=Depends(get_current_worker),
):
    result = service.mark_read(db, reminder_id, worker.id)
    return ok(result, "已标记为已读")

File: week01/day07/frontend/src/services/reminder.ts
import { request } from './request';
import type { PageResult } from '../types/api';
import type { Reminder } from '../types/reminder';

export function listReminders(page = 1, pageSize = 20): Promise<PageResult<Reminder>> {
  return request<PageResult<Reminder>>({
    method: 'GET',
    url: '/api/reminders',
    params: { page, pageSize },
  });
}

export function markRead(id: number): Promise<unknown> {
  return request<unknown>({
    method: 'PATCH',
    url: `/api/reminders/${id}/read`,
  });
}
